File: sesion-20/mundo_grafico.cpp
// =============================================================
//  PRACTICA 5 · Diseno modular, funciones y aplicaciones graficas
//  "Mundo-grafico" · aplicacion modular de figuras geometricas
//  Fundamentos de Programacion · UNITEC Atizapan · Ciclo 26-3
// =============================================================
//  Este archivo es un PLANO: trae los pasos y las firmas de las
//  funciones comentadas. Escribe TU el cuerpo de cada funcion.
//  No borres los pasos: al final quedaran como los comentarios
//  que documentan tu programa.
//
//  Meta: dibujar figuras (circulo, rectangulo, linea) con
//  funciones modulares y luego moverlas / cambiarlas de tamano.
//  Software oficial: Dev C++ con la libreria graphics.h (WinBGIm).
//
//  Recuerda los tres modulos que acordamos en clase:
//    - MODULO DE FIGURAS   -> crea y dibuja las figuras
//    - OPERACIONES         -> mueve y cambia el tamano
//    - MODULO PRINCIPAL    -> main(): coordina e interactua
// =============================================================

// PASO 1: incluye las librerias
//         #include <graphics.h>   // dibujo: circle, rectangle, line, setcolor
//         #include <iostream>

using namespace std;

// ---------- MODULO DE FIGURAS (funciones con VARIOS parametros) ----------

// PASO 2: void dibujar_circulo(int x, int y, int radio, int color)
//   Funcion con VARIOS parametros. En el cuerpo:
//     setcolor(color);        // color del trazo
//     circle(x, y, radio);    // dibuja el circulo
//   No devuelve nada -> por eso es void.

// PASO 3: void dibujar_rectangulo(int x, int y, int ancho, int alto, int color)
//   setcolor(color);
//   rectangle(x, y, x + ancho, y + alto);

// PASO 4: void dibujar_linea(int x1, int y1, int x2, int y2, int color)
//   setcolor(color);
//   line(x1, y1, x2, y2);

// ---------- OPERACIONES (DEVOLUCION del resultado con return) ----------

// PASO 5: int mover_derecha(int x, int paso)
//   Funcion que RECIBE la posicion y el paso, y DEVUELVE la nueva
//   posicion:  return x + paso;

// PASO 6: int cambiar_tamano(int radio, int cuanto)
//   Devuelve el nuevo radio:  return radio + cuanto;

// ---------- MODULO PRINCIPAL (coordina las piezas) ----------

int main() {

    // PASO 7: abre la ventana grafica
    //         initwindow(600, 400, "Mundo-grafico");

    // PASO 8: dibuja UNA escena llamando a TUS funciones del modulo
    //         de figuras (elige posiciones, tamanos y colores):
    //         dibujar_circulo(...);  dibujar_rectangulo(...);  dibujar_linea(...);

    // PASO 9: ANIMACION BASICA. Un ciclo (por ejemplo 60 vueltas) que:
    //         1) borra la pantalla        -> cleardevice();
    //         2) dibuja el circulo en x   -> dibujar_circulo(x, y, radio, color);
    //         3) espera un momento        -> delay(30);
    //         4) mueve la figura          -> x = mover_derecha(x, 5);

    // PASO 10: espera una tecla y cierra la ventana
    //          getch();  closegraph();

    return 0;
}

/* CONCLUSIONES:
   ¿Para que sirve el diseno modular en la programacion?
   Escribe aqui una redaccion breve con tus observaciones sobre
   los resultados y los conocimientos adquiridos al concluir la
   practica. */

File: sesion-20/mundo_grafico.py
# =============================================================
#  PRÁCTICA 5 · Diseño modular, funciones y aplicaciones gráficas
#  "Mundo-gráfico" · aplicación modular de figuras geométricas
#  Fundamentos de Programación · UNITEC Atizapán · Ciclo 26-3
# =============================================================
#  Este archivo es un PLANO: trae los pasos y las firmas de las
#  funciones comentadas. Escribe TÚ el cuerpo de cada función.
#
#  Nota: C++ usa la librería graphics.h (Dev C++). En Python el
#  puente más cercano es turtle, que ya viene incluido y dibuja
#  círculos, rectángulos y líneas con posición, tamaño y color.
#
#  Los tres módulos son los mismos:
#    - MÓDULO DE FIGURAS   -> crea y dibuja las figuras
#    - OPERACIONES         -> mueve y cambia el tamaño
#    - MÓDULO PRINCIPAL    -> coordina e interactúa
# =============================================================

import turtle

# PASO 1: crea la tortuga y la pantalla
#         t = turtle.Turtle()
#         pantalla = turtle.Screen()

# ---------- MÓDULO DE FIGURAS (funciones con VARIOS parámetros) ----------

# PASO 2: def dibujar_circulo(x, y, radio, color):
#   Función con VARIOS parámetros. En el cuerpo:
#     t.penup(); t.goto(x, y); t.pendown()   # coloca sin dibujar
#     t.color(color)                         # color del trazo
#     t.circle(radio)                        # dibuja el círculo

# PASO 3: def dibujar_rectangulo(x, y, ancho, alto, color):
#   Coloca la tortuga en (x, y) y dibuja los 4 lados con un for
#   usando t.forward(...) y t.left(90).

# PASO 4: def dibujar_linea(x1, y1, x2, y2, color):
#   t.penup(); t.goto(x1, y1); t.pendown(); t.goto(x2, y2)

# ---------- OPERACIONES (DEVOLUCIÓN del resultado con return) ----------

# PASO 5: def mover_derecha(x, paso):
#   Recibe la posición y el paso y DEVUELVE la nueva posición:
#       return x + paso

# PASO 6: def cambiar_tamano(radio, cuanto):
#       return radio + cuanto

# ---------- MÓDULO PRINCIPAL (coordina las piezas) ----------

# PASO 7: dibuja UNA escena llamando a TUS funciones del módulo
#         de figuras (posiciones, tamaños y colores a tu gusto).

# PASO 8: ANIMACIÓN BÁSICA. Un ciclo que:
#         1) borra           -> t.clear()
#         2) dibuja el círculo en x
#         3) mueve la figura -> x = mover_derecha(x, 5)

# PASO 9: mantén la ventana abierta -> pantalla.mainloop()

# CONCLUSIONES:
# ¿Para qué sirve el diseño modular en la programación?
# Escribe aquí una redacción breve con tus observaciones sobre
# los resultados y los conocimientos adquiridos al concluir la
# práctica.
